[BUG] CMake: cannot create ALIAS target "SUNDIALS::KLU" because target "SuiteSparse::KLU" is itself an ALIAS

### Current Behavior:
Configuring sundials with KLU fails with

```
  -- KLU version: 2.3.4
  -- KLU include: /home/dweindl/src/AMICI-devel/python/sdist/amici/include/suitesparse
  -- KLU library: /home/dweindl/src/AMICI-devel/python/sdist/amici/lib/libklu.a
  -- KLU static:  /home/dweindl/src/AMICI-devel/python/sdist/amici/lib/libklu.a
  CMake Error at cmake/tpl/FindKLU.cmake:40 (add_library):
    add_library cannot create ALIAS target "SUNDIALS::KLU" because target
    "SuiteSparse::KLU" is itself an ALIAS.
  Call Stack (most recent call first):
    cmake/tpl/SundialsKLU.cmake:48 (find_package)
    cmake/SundialsSetupTPLs.cmake:113 (include)
    CMakeLists.txt:186 (include)
```

### Expected Behavior:

Configuring succeeds.

### Steps To Reproduce:

* Build SuiteSparse 7.8.2 with `BUILD_SHARED_LIBS=OFF`
* Set `CMAKE_PREFIX_PATH` so the SuiteSparse CMake config can be found
* Configure sundials with `-DENABLE_KLU=ON` (and not setting any `KLU_*`)

### Environment:
- SUNDIALS: 7.1.1
- SuiteSparse: 7.8.2
- OS: Ubuntu 24.04
- CMake: 3.30.3


### Anything else:

Here, sundials tries to create an ALIAS target `SUNDIALS::KLU` for `SuiteSparse::KLU`:

https://github.com/LLNL/sundials/blob/c28eaa3764a03705d61decb6025b409360e9d53f/cmake/tpl/FindKLU.cmake#L33-L40

This works as long as `SuiteSparse::KLU` is not an alias target itself.

However, for a SuiteSparse static-only build, `SuiteSparse::KLU` is already an ALIAS of `SuiteSparse::KLU_static`:

https://github.com/DrTimothyAldenDavis/SuiteSparse/blob/c8c3a9de1c8eef54da5ff19fd0bcf7ca6e8bc9de/KLU/Config/KLUConfig.cmake.in#L113-L116

### Proposed fix:

Replace

https://github.com/LLNL/sundials/blob/c28eaa3764a03705d61decb6025b409360e9d53f/cmake/tpl/FindKLU.cmake#L40

by

```cmake
      get_target_property(klu_aliased_target SuiteSparse::KLU ALIASED_TARGET)
      if (klu_aliased_target)
          add_library(SUNDIALS::KLU ALIAS ${klu_aliased_target})
      else()
          add_library(SUNDIALS::KLU ALIAS SuiteSparse::KLU)
      endif()
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -39,6 +39,9 @@ repeatedly.
 Fixed compilation errors when building the Trilinos Teptra NVector with CUDA
 support.
 
+Fixed a CMake configuration issue related to aliasing an `ALIAS` target when
+using `ENABLE_KLU=ON` in combination with a static-only build of SuiteSparse.
+
 ### Deprecation Notices
 
 ## Changes to SUNDIALS in release 7.1.1

--- a/cmake/tpl/FindKLU.cmake
+++ b/cmake/tpl/FindKLU.cmake
@@ -43,7 +43,7 @@ if(NOT
       # For static-only builds of SuiteSparse, SuiteSparse::KLU will itself be
       # an ALIAS target which can't be aliased.
       get_target_property(klu_aliased_target SuiteSparse::KLU ALIASED_TARGET)
-      if (klu_aliased_target)
+      if(klu_aliased_target)
         add_library(SUNDIALS::KLU ALIAS ${klu_aliased_target})
       else()
         add_library(SUNDIALS::KLU ALIAS SuiteSparse::KLU)

--- a/cmake/tpl/FindKLU.cmake
+++ b/cmake/tpl/FindKLU.cmake
@@ -48,7 +48,7 @@ if(NOT
       else()
         add_library(SUNDIALS::KLU ALIAS SuiteSparse::KLU)
       endif()
-    set(KLU_SUITESPARSE_TARGET ON)
+      set(KLU_SUITESPARSE_TARGET ON)
       mark_as_advanced(KLU_SUITESPARSE_TARGET)
     endif()
     return()

--- a/cmake/tpl/FindKLU.cmake
+++ b/cmake/tpl/FindKLU.cmake
@@ -40,8 +40,15 @@ if(NOT
 
   if(TARGET SuiteSparse::KLU)
     if(NOT TARGET SUNDIALS::KLU)
-      add_library(SUNDIALS::KLU ALIAS SuiteSparse::KLU)
-      set(KLU_SUITESPARSE_TARGET ON)
+      # For static-only builds of SuiteSparse, SuiteSparse::KLU will itself be
+      # an ALIAS target which can't be aliased.
+      get_target_property(klu_aliased_target SuiteSparse::KLU ALIASED_TARGET)
+      if (klu_aliased_target)
+        add_library(SUNDIALS::KLU ALIAS ${klu_aliased_target})
+      else()
+        add_library(SUNDIALS::KLU ALIAS SuiteSparse::KLU)
+      endif()
+    set(KLU_SUITESPARSE_TARGET ON)
       mark_as_advanced(KLU_SUITESPARSE_TARGET)
     endif()
     return()

--- a/doc/shared/RecentChanges.rst
+++ b/doc/shared/RecentChanges.rst
@@ -35,4 +35,7 @@ repeatedly.
 Fixed compilation errors when building the Trilinos Teptra NVector with CUDA
 support.
 
+Fixed a CMake configuration issue related to aliasing an ``ALIAS`` target when
+using ``ENABLE_KLU=ON`` in combination with a static-only build of SuiteSparse.
+
 **Deprecation Notices**

